### PR TITLE
appliance: deploy redis services

### PR DESCRIPTION
**Pull request by craigfurman**
**appliance: delete redis-exporter config**

**appliance: remove unused extra env vars config fields**

**appliance/compare-helm: support multiple components**

Helps with redis services, which have helm templates with component
names including redis, redis-cache, redis-store.

**appliance: deploy redis services**

---

Closes https://github.com/sourcegraph/sourcegraph-operator/issues/98

## Test plan

Golden tests included

### CODE REVIEW

**Comment by craigfurman** [comment]
Annotated output of:

```
go run ./internal/appliance/dev/compare-helm \
  -component redis,redis-store,redis-cache \
  -golden-file internal/appliance/testdata/golden-fixtures/redis/default.yaml
```

```diff
2c2
< # helm: Secret/redis-cache
---
> # golden: Secret/redis-cache
7a8,10
>   annotations:
>     appliance.sourcegraph.com/configHash: 142c00ae31a1d919c6bebf39a02cfd56ecf57e219dc5427b0fbeba316a831ebf
>   creationTimestamp: "2024-04-19T00:00:00Z"
9d11
<     app: redis-cache
10a13,14
>     app.kubernetes.io/name: sourcegraph
>     app.kubernetes.io/version: 5.3.9104
12a17,26
>   namespace: NORMALIZED_FOR_TESTING
>   ownerReferences:
>   - apiVersion: v1
>     blockOwnerDeletion: true
>     controller: true
>     kind: ConfigMap
>     name: sg
>     uid: NORMALIZED_FOR_TESTING
>   resourceVersion: NORMALIZED_FOR_TESTING
>   uid: NORMALIZED_FOR_TESTING
15c29
< # helm: Secret/redis-store
---
> # golden: Secret/redis-store
20a35,37
>   annotations:
>     appliance.sourcegraph.com/configHash: 142c00ae31a1d919c6bebf39a02cfd56ecf57e219dc5427b0fbeba316a831ebf
>   creationTimestamp: "2024-04-19T00:00:00Z"
22d38
<     app: redis-store
23a40,41
>     app.kubernetes.io/name: sourcegraph
>     app.kubernetes.io/version: 5.3.9104
25a44,53
>   namespace: NORMALIZED_FOR_TESTING
>   ownerReferences:
>   - apiVersion: v1
>     blockOwnerDeletion: true
>     controller: true
>     kind: ConfigMap
>     name: sg
>     uid: NORMALIZED_FOR_TESTING
>   resourceVersion: NORMALIZED_FOR_TESTING
>   uid: NORMALIZED_FOR_TESTING
28c56
< # helm: PersistentVolumeClaim/redis-cache
---
> # golden: PersistentVolumeClaim/redis-cache
31a60,64
>   annotations:
>     appliance.sourcegraph.com/configHash: 142c00ae31a1d919c6bebf39a02cfd56ecf57e219dc5427b0fbeba316a831ebf
>   creationTimestamp: "2024-04-19T00:00:00Z"
>   finalizers:
>   - kubernetes.io/pvc-protection
33d65
<     app.kubernetes.io/component: redis
35a68,77
>   namespace: NORMALIZED_FOR_TESTING
>   ownerReferences:
>   - apiVersion: v1
>     blockOwnerDeletion: true
>     controller: true
>     kind: ConfigMap
>     name: sg
>     uid: NORMALIZED_FOR_TESTING
>   resourceVersion: NORMALIZED_FOR_TESTING
>   uid: NORMALIZED_FOR_TESTING
42a85,87
>   volumeMode: Filesystem
> status:
>   phase: Pending
44c89
< # helm: PersistentVolumeClaim/redis-store
---
> # golden: PersistentVolumeClaim/redis-store
47a93,97
>   annotations:
>     appliance.sourcegraph.com/configHash: 142c00ae31a1d919c6bebf39a02cfd56ecf57e219dc5427b0fbeba316a831ebf
>   creationTimestamp: "2024-04-19T00:00:00Z"
>   finalizers:
>   - kubernetes.io/pvc-protection
49d98
<     app.kubernetes.io/component: redis
51a101,110
>   namespace: NORMALIZED_FOR_TESTING
>   ownerReferences:
>   - apiVersion: v1
>     blockOwnerDeletion: true
>     controller: true
>     kind: ConfigMap
>     name: sg
>     uid: NORMALIZED_FOR_TESTING
>   resourceVersion: NORMALIZED_FOR_TESTING
>   uid: NORMALIZED_FOR_TESTING
58a118,120
>   volumeMode: Filesystem
> status:
>   phase: Pending
60c122
< # helm: Service/redis-cache
---
> # golden: Service/redis-cache
64a127
>     appliance.sourcegraph.com/configHash: 142c00ae31a1d919c6bebf39a02cfd56ecf57e219dc5427b0fbeba316a831ebf
66a130
>   creationTimestamp: "2024-04-19T00:00:00Z"
69c133
<     app.kubernetes.io/component: redis
---
>     app.kubernetes.io/component: redis-cache
71a136,145
>   namespace: NORMALIZED_FOR_TESTING
>   ownerReferences:
>   - apiVersion: v1
>     blockOwnerDeletion: true
>     controller: true
>     kind: ConfigMap
>     name: sg
>     uid: NORMALIZED_FOR_TESTING
>   resourceVersion: NORMALIZED_FOR_TESTING
>   uid: NORMALIZED_FOR_TESTING
72a147,153
>   clusterIP: NORMALIZED_FOR_TESTING
>   clusterIPs:
>   - NORMALIZED_FOR_TESTING
>   internalTrafficPolicy: Cluster
>   ipFamilies:
>   - IPv4
>   ipFamilyPolicy: SingleStack
75a157
>     protocol: TCP
79,80c161
<     app.kubernetes.io/instance: release-name
<     app.kubernetes.io/name: sourcegraph
---
>   sessionAffinity: None
81a163,164
> status:
>   loadBalancer: {}
83c166
< # helm: Service/redis-store
---
> # golden: Service/redis-store
87a171
>     appliance.sourcegraph.com/configHash: 142c00ae31a1d919c6bebf39a02cfd56ecf57e219dc5427b0fbeba316a831ebf
89a174
>   creationTimestamp: "2024-04-19T00:00:00Z"
92c177
<     app.kubernetes.io/component: redis
---
>     app.kubernetes.io/component: redis-store
94a180,189
>   namespace: NORMALIZED_FOR_TESTING
>   ownerReferences:
>   - apiVersion: v1
>     blockOwnerDeletion: true
>     controller: true
>     kind: ConfigMap
>     name: sg
>     uid: NORMALIZED_FOR_TESTING
>   resourceVersion: NORMALIZED_FOR_TESTING
>   uid: NORMALIZED_FOR_TESTING
95a191,197
>   clusterIP: NORMALIZED_FOR_TESTING
>   clusterIPs:
>   - NORMALIZED_FOR_TESTING
>   internalTrafficPolicy: Cluster
>   ipFamilies:
>   - IPv4
>   ipFamilyPolicy: SingleStack
98a201
>     protocol: TCP
102,103c205
<     app.kubernetes.io/instance: release-name
<     app.kubernetes.io/name: sourcegraph
---
>   sessionAffinity: None
104a207,208
> status:
>   loadBalancer: {}
106c210
< # helm: Deployment/redis-cache
---
> # golden: Deployment/redis-cache
111c215,217
<     description: Redis for storing short-lived caches.
---
>     appliance.sourcegraph.com/configHash: 142c00ae31a1d919c6bebf39a02cfd56ecf57e219dc5427b0fbeba316a831ebf
>   creationTimestamp: "2024-04-19T00:00:00Z"
>   generation: 1
113,115c219
<     app.kubernetes.io/component: redis
<     app.kubernetes.io/instance: release-name
<     app.kubernetes.io/managed-by: Helm
---
>     app.kubernetes.io/component: redis-cache
117c221
<     app.kubernetes.io/version: 5.3.2
---
>     app.kubernetes.io/version: 5.3.9104
119d222
<     helm.sh/chart: sourcegraph-5.3.2
120a224,233
>   namespace: NORMALIZED_FOR_TESTING
>   ownerReferences:
>   - apiVersion: v1
>     blockOwnerDeletion: true
>     controller: true
>     kind: ConfigMap
>     name: sg
>     uid: NORMALIZED_FOR_TESTING
>   resourceVersion: NORMALIZED_FOR_TESTING
>   uid: NORMALIZED_FOR_TESTING
122a236
>   progressDeadlineSeconds: 600
128,129d241
<       app.kubernetes.io/instance: release-name
<       app.kubernetes.io/name: sourcegraph
135a248
>       creationTimestamp: null
138,139d250
<         app.kubernetes.io/instance: release-name
<         app.kubernetes.io/name: sourcegraph
140a252
>       name: redis-cache
142d253
<       affinity: null
144,145c255
<       - env: null
<         image: index.docker.io/sourcegraph/redis-cache:5.3.2@sha256:ed79dada4d1a2bd85fb8450dffe227283ab6ae0e7ce56dc5056fbb8202d95624
---
>       - image: index.docker.io/sourcegraph/redis-cache:5.3.2@sha256:ed79dada4d1a2bd85fb8450dffe227283ab6ae0e7ce56dc5056fbb8202d95624
150a261
>           successThreshold: 1
157a269
>           protocol: TCP
163,174c275,279
<             - |
<               #!/bin/bash
<               if [ -f /etc/redis/redis.conf ]; then
<                 REDISCLI_AUTH=$(grep -h "requirepass" /etc/redis/redis.conf | cut -d ' ' -f 2)
<               fi
<               response=$(
<                 redis-cli ping
<               )
<               if [ "$response" != "PONG" ]; then
<                 echo "$response"
<                 exit 1
<               fi
---
>             - "\n#!/bin/bash\nif [ -f /etc/redis/redis.conf ]; then\n  REDISCLI_AUTH=$(grep
>               -h \"requirepass\" /etc/redis/redis.conf | cut -d ' ' -f 2)\nfi\nresponse=$(\n
>               \ redis-cli ping\n)\nif [ \"$response\" != \"PONG\" ]; then\n  echo
>               \"$response\"\n  exit 1\nfi\n\t\t\t\t\t"
>           failureThreshold: 3
175a281,283
>           periodSeconds: 10
>           successThreshold: 1
>           timeoutSeconds: 1
187a296
>         terminationMessagePath: /dev/termination-log
192,193c301
<       - env: null
<         image: index.docker.io/sourcegraph/redis_exporter:5.3.2@sha256:21a9dd9214483a42b11d58bf99e4f268f44257a4f67acd436d458797a31b7786
---
>       - image: index.docker.io/sourcegraph/redis_exporter:5.3.2@sha256:21a9dd9214483a42b11d58bf99e4f268f44257a4f67acd436d458797a31b7786
198a307
>           protocol: TCP
210a320
>         terminationMessagePath: /dev/termination-log
212c322,324
<       nodeSelector: null
---
>       dnsPolicy: ClusterFirst
>       restartPolicy: Always
>       schedulerName: default-scheduler
216c328,330
<       tolerations: null
---
>         runAsGroup: 101
>         runAsUser: 100
>       terminationGracePeriodSeconds: 30
220a335
> status: {}
222c337
< # helm: Deployment/redis-store
---
> # golden: Deployment/redis-store
227c342,344
<     description: Redis for storing semi-persistent data like user sessions.
---
>     appliance.sourcegraph.com/configHash: 142c00ae31a1d919c6bebf39a02cfd56ecf57e219dc5427b0fbeba316a831ebf
>   creationTimestamp: "2024-04-19T00:00:00Z"
>   generation: 1
229,231c346
<     app.kubernetes.io/component: redis
<     app.kubernetes.io/instance: release-name
<     app.kubernetes.io/managed-by: Helm
---
>     app.kubernetes.io/component: redis-store
233c348
<     app.kubernetes.io/version: 5.3.2
---
>     app.kubernetes.io/version: 5.3.9104
235d349
<     helm.sh/chart: sourcegraph-5.3.2
236a351,360
>   namespace: NORMALIZED_FOR_TESTING
>   ownerReferences:
>   - apiVersion: v1
>     blockOwnerDeletion: true
>     controller: true
>     kind: ConfigMap
>     name: sg
>     uid: NORMALIZED_FOR_TESTING
>   resourceVersion: NORMALIZED_FOR_TESTING
>   uid: NORMALIZED_FOR_TESTING
238a363
>   progressDeadlineSeconds: 600
244,245d368
<       app.kubernetes.io/instance: release-name
<       app.kubernetes.io/name: sourcegraph
251a375
>       creationTimestamp: null
254,255d377
<         app.kubernetes.io/instance: release-name
<         app.kubernetes.io/name: sourcegraph
256a379
>       name: redis-store
258d380
<       affinity: null
260,261c382
<       - env: null
<         image: index.docker.io/sourcegraph/redis-store:5.3.2@sha256:0e3270a5eb293c158093f41145810eb5a154f61a74c9a896690dfdecd1b98b39
---
>       - image: index.docker.io/sourcegraph/redis-store:5.3.2@sha256:0e3270a5eb293c158093f41145810eb5a154f61a74c9a896690dfdecd1b98b39
266a388
>           successThreshold: 1
273a396
>           protocol: TCP
279,290c402,406
<             - |
<               #!/bin/bash
<               if [ -f /etc/redis/redis.conf ]; then
<                 REDISCLI_AUTH=$(grep -h "requirepass" /etc/redis/redis.conf | cut -d ' ' -f 2)
<               fi
<               response=$(
<                 redis-cli ping
<               )
<               if [ "$response" != "PONG" ]; then
<                 echo "$response"
<                 exit 1
<               fi
---
>             - "\n#!/bin/bash\nif [ -f /etc/redis/redis.conf ]; then\n  REDISCLI_AUTH=$(grep
>               -h \"requirepass\" /etc/redis/redis.conf | cut -d ' ' -f 2)\nfi\nresponse=$(\n
>               \ redis-cli ping\n)\nif [ \"$response\" != \"PONG\" ]; then\n  echo
>               \"$response\"\n  exit 1\nfi\n\t\t\t\t\t"
>           failureThreshold: 3
291a408,410
>           periodSeconds: 10
>           successThreshold: 1
>           timeoutSeconds: 1
303a423
>         terminationMessagePath: /dev/termination-log
308,309c428
<       - env: null
<         image: index.docker.io/sourcegraph/redis_exporter:5.3.2@sha256:21a9dd9214483a42b11d58bf99e4f268f44257a4f67acd436d458797a31b7786
---
>       - image: index.docker.io/sourcegraph/redis_exporter:5.3.2@sha256:21a9dd9214483a42b11d58bf99e4f268f44257a4f67acd436d458797a31b7786
314a434
>           protocol: TCP
326a447
>         terminationMessagePath: /dev/termination-log
328c449,451
<       nodeSelector: null
---
>       dnsPolicy: ClusterFirst
>       restartPolicy: Always
>       schedulerName: default-scheduler
332c455,457
<       tolerations: null
---
>         runAsGroup: 101
>         runAsUser: 100
>       terminationGracePeriodSeconds: 30
336a462,528
> status: {}
> ---
> # golden: ConfigMap/sg
> apiVersion: v1
> data:
>   spec: |
>     spec:
>       requestedVersion: "5.3.9104"
> 
>       blobstore:
>         disabled: true
> 
>       codeInsights:
>         disabled: true
> 
>       codeIntel:
>         disabled: true
> 
>       frontend:
>         disabled: true
> 
>       gitServer:
>         disabled: true
> 
>       indexedSearch:
>         disabled: true
> 
>       indexedSearchIndexer:
>         disabled: true
> 
>       pgsql:
>         disabled: true
> 
>       postgresExporter:
>         disabled: true
> 
>       preciseCodeIntel:
>         disabled: true
> 
>       redisCache: {}
> 
>       redisStore: {}
> 
>       repoUpdater:
>         disabled: true
> 
>       searcher:
>         disabled: true
> 
>       symbols:
>         disabled: true
> 
>       syntectServer:
>         disabled: true
> 
>       worker:
>         disabled: true
> kind: ConfigMap
> metadata:
>   annotations:
>     appliance.sourcegraph.com/currentVersion: 5.3.9104
>     appliance.sourcegraph.com/managed: "true"
>   creationTimestamp: "2024-04-19T00:00:00Z"
>   name: sg
>   namespace: NORMALIZED_FOR_TESTING
>   resourceVersion: NORMALIZED_FOR_TESTING
>   uid: NORMALIZED_FOR_TESTING
```

File: internal/appliance/spec.go [hunk 1]
```diff
@@ -196,66 +169,20 @@ type PreciseCodeIntelSpec struct {
...
-}
-
-// RedisCacheSpec defines the desired state of the Redis cache service.
-type RedisCacheSpec struct {
-	// Disabled defines if Redis Cache is enabled or not.
-	// Default: false
-	Disabled bool `json:"disabled,omitempty"`
-
-	// ExistingSecret is the name of an existing secret to use.
-	ExistingSecret string `json:"existingSecret,omitempty"`
```
Comment (craigfurman) [inline internal/appliance/spec.go:211]: The helm chart has the equivalent field as configurable. AFAICT, it does not make sense to set it there unless the redis instance (cache / store) is _disabled_. The user is injecting details of an external redis, so they probably don't want SG to provision one too!

I think there's a case for _not_ making this field configurable here. The user can just disable redis-cache and/or redis-store, and create secret(s) with name "redis-cache" and/or redis-store in SG's namespace.

It's certainly not a big deal to make this configurable, but we will have to do a bit of a refactor, pushing more of spec.go into the config subpackage, in order to import it from container.EnvVarsRedis().

WDYT @jdpleiness?

File: internal/appliance/spec.go [hunk 2]
```diff
@@ -196,66 +169,20 @@ type PreciseCodeIntelSpec struct {
...
-
-	// Resources allows for custom resource limits and requests.
-	Resources *corev1.ResourceList `json:"resources,omitempty"`
-
-	// Env defines environment variables for Redis Cache.
-	Env map[string]string `json:"env,omitempty"`
 }
 
-// RedisExporterSpec defines the desired state of the Redis Exporter service.
-type RedisExporterSpec struct {
```
Comment (craigfurman) [inline internal/appliance/spec.go:225]: self TODO: we should be able to make this container's resources overridable with standard config in the cache/store specs. Write a test for this.

**Review by craigfurman — COMMENTED** [review]
@jdpleiness 2 questions for you. I'm not 100% happy with the compare-helm diff yet (see above), and will take a look at that tomorrow, so no need for a full review unless you're really curious 😁

File: internal/appliance/testdata/golden-fixtures/redis/default.yaml [hunk 3]
```diff
@@ -0,0 +1,511 @@
+resources:
```
Comment (craigfurman) [inline internal/appliance/testdata/golden-fixtures/redis/default.yaml:1]: Annotated output of

```
go run ./internal/appliance/dev/compare-helm \
  -component redis,redis-store,redis-cache \
  -golden-file internal/appliance/testdata/golden-fixtures/redis/default.yaml

```

I don't see anything that looks problematic in there, but more eyes very welcome!

```diff
2c2
< # helm: Secret/redis-cache
---
> # golden: Secret/redis-cache
7a8,10
>   annotations:
>     appliance.sourcegraph.com/configHash: 142c00ae31a1d919c6bebf39a02cfd56ecf57e219dc5427b0fbeba316a831ebf
>   creationTimestamp: "2024-04-19T00:00:00Z"
9d11
<     app: redis-cache
10a13,14
>     app.kubernetes.io/name: sourcegraph
>     app.kubernetes.io/version: 5.3.9104
12a17,26
>   namespace: NORMALIZED_FOR_TESTING
>   ownerReferences:
>   - apiVersion: v1
>     blockOwnerDeletion: true
>     controller: true
>     kind: ConfigMap
>     name: sg
>     uid: NORMALIZED_FOR_TESTING
>   resourceVersion: NORMALIZED_FOR_TESTING
>   uid: NORMALIZED_FOR_TESTING
15c29
< # helm: Secret/redis-store
---
> # golden: Secret/redis-store
20a35,37
>   annotations:
>     appliance.sourcegraph.com/configHash: 142c00ae31a1d919c6bebf39a02cfd56ecf57e219dc5427b0fbeba316a831ebf
>   creationTimestamp: "2024-04-19T00:00:00Z"
22d38
<     app: redis-store
23a40,41
>     app.kubernetes.io/name: sourcegraph
>     app.kubernetes.io/version: 5.3.9104
25a44,53
>   namespace: NORMALIZED_FOR_TESTING
>   ownerReferences:
>   - apiVersion: v1
>     blockOwnerDeletion: true
>     controller: true
>     kind: ConfigMap
>     name: sg
>     uid: NORMALIZED_FOR_TESTING
>   resourceVersion: NORMALIZED_FOR_TESTING
>   uid: NORMALIZED_FOR_TESTING
28c56
< # helm: PersistentVolumeClaim/redis-cache
---
> # golden: PersistentVolumeClaim/redis-cache
31a60,64
>   annotations:
>     appliance.sourcegraph.com/configHash: 142c00ae31a1d919c6bebf39a02cfd56ecf57e219dc5427b0fbeba316a831ebf
>   creationTimestamp: "2024-04-19T00:00:00Z"
>   finalizers:
>   - kubernetes.io/pvc-protection
33d65
<     app.kubernetes.io/component: redis
35a68,77
>   namespace: NORMALIZED_FOR_TESTING
>   ownerReferences:
>   - apiVersion: v1
>     blockOwnerDeletion: true
>     controller: true
>     kind: ConfigMap
>     name: sg
>     uid: NORMALIZED_FOR_TESTING
>   resourceVersion: NORMALIZED_FOR_TESTING
>   uid: NORMALIZED_FOR_TESTING
42a85,87
>   volumeMode: Filesystem
> status:
>   phase: Pending
44c89
< # helm: PersistentVolumeClaim/redis-store
---
> # golden: PersistentVolumeClaim/redis-store
47a93,97
>   annotations:
>     appliance.sourcegraph.com/configHash: 142c00ae31a1d919c6bebf39a02cfd56ecf57e219dc5427b0fbeba316a831ebf
>   creationTimestamp: "2024-04-19T00:00:00Z"
>   finalizers:
>   - kubernetes.io/pvc-protection
49d98
<     app.kubernetes.io/component: redis
51a101,110
>   namespace: NORMALIZED_FOR_TESTING
>   ownerReferences:
>   - apiVersion: v1
>     blockOwnerDeletion: true
>     controller: true
>     kind: ConfigMap
>     name: sg
>     uid: NORMALIZED_FOR_TESTING
>   resourceVersion: NORMALIZED_FOR_TESTING
>   uid: NORMALIZED_FOR_TESTING
58a118,120
>   volumeMode: Filesystem
> status:
>   phase: Pending
60c122
< # helm: Service/redis-cache
---
> # golden: Service/redis-cache
64a127
>     appliance.sourcegraph.com/configHash: 142c00ae31a1d919c6bebf39a02cfd56ecf57e219dc5427b0fbeba316a831ebf
66a130
>   creationTimestamp: "2024-04-19T00:00:00Z"
69c133
<     app.kubernetes.io/component: redis
---
>     app.kubernetes.io/component: redis-cache
71a136,145
>   namespace: NORMALIZED_FOR_TESTING
>   ownerReferences:
>   - apiVersion: v1
>     blockOwnerDeletion: true
>     controller: true
>     kind: ConfigMap
>     name: sg
>     uid: NORMALIZED_FOR_TESTING
>   resourceVersion: NORMALIZED_FOR_TESTING
>   uid: NORMALIZED_FOR_TESTING
72a147,153
>   clusterIP: NORMALIZED_FOR_TESTING
>   clusterIPs:
>   - NORMALIZED_FOR_TESTING
>   internalTrafficPolicy: Cluster
>   ipFamilies:
>   - IPv4
>   ipFamilyPolicy: SingleStack
75a157
>     protocol: TCP
79,80c161
<     app.kubernetes.io/instance: release-name
<     app.kubernetes.io/name: sourcegraph
---
>   sessionAffinity: None
81a163,164
> status:
>   loadBalancer: {}
83c166
< # helm: Service/redis-store
---
> # golden: Service/redis-store
87a171
>     appliance.sourcegraph.com/configHash: 142c00ae31a1d919c6bebf39a02cfd56ecf57e219dc5427b0fbeba316a831ebf
89a174
>   creationTimestamp: "2024-04-19T00:00:00Z"
92c177
<     app.kubernetes.io/component: redis
---
>     app.kubernetes.io/component: redis-store
94a180,189
>   namespace: NORMALIZED_FOR_TESTING
>   ownerReferences:
>   - apiVersion: v1
>     blockOwnerDeletion: true
>     controller: true
>     kind: ConfigMap
>     name: sg
>     uid: NORMALIZED_FOR_TESTING
>   resourceVersion: NORMALIZED_FOR_TESTING
>   uid: NORMALIZED_FOR_TESTING
95a191,197
>   clusterIP: NORMALIZED_FOR_TESTING
>   clusterIPs:
>   - NORMALIZED_FOR_TESTING
>   internalTrafficPolicy: Cluster
>   ipFamilies:
>   - IPv4
>   ipFamilyPolicy: SingleStack
98a201
>     protocol: TCP
102,103c205
<     app.kubernetes.io/instance: release-name
<     app.kubernetes.io/name: sourcegraph
---
>   sessionAffinity: None
104a207,208
> status:
>   loadBalancer: {}
106c210
< # helm: Deployment/redis-cache
---
> # golden: Deployment/redis-cache
111c215,217
<     description: Redis for storing short-lived caches.
---
>     appliance.sourcegraph.com/configHash: 142c00ae31a1d919c6bebf39a02cfd56ecf57e219dc5427b0fbeba316a831ebf
>   creationTimestamp: "2024-04-19T00:00:00Z"
>   generation: 1
113,115c219
<     app.kubernetes.io/component: redis
<     app.kubernetes.io/instance: release-name
<     app.kubernetes.io/managed-by: Helm
---
>     app.kubernetes.io/component: redis-cache
117c221
<     app.kubernetes.io/version: 5.3.2
---
>     app.kubernetes.io/version: 5.3.9104
119d222
<     helm.sh/chart: sourcegraph-5.3.2
120a224,233
>   namespace: NORMALIZED_FOR_TESTING
>   ownerReferences:
>   - apiVersion: v1
>     blockOwnerDeletion: true
>     controller: true
>     kind: ConfigMap
>     name: sg
>     uid: NORMALIZED_FOR_TESTING
>   resourceVersion: NORMALIZED_FOR_TESTING
>   uid: NORMALIZED_FOR_TESTING
122a236
>   progressDeadlineSeconds: 600
128,129d241
<       app.kubernetes.io/instance: release-name
<       app.kubernetes.io/name: sourcegraph
135a248
>       creationTimestamp: null
138,139d250
<         app.kubernetes.io/instance: release-name
<         app.kubernetes.io/name: sourcegraph
140a252
>       name: redis-cache
142d253
<       affinity: null
144,145c255
<       - env: null
<         image: index.docker.io/sourcegraph/redis-cache:5.3.2@sha256:ed79dada4d1a2bd85fb8450dffe227283ab6ae0e7ce56dc5056fbb8202d95624
---
>       - image: index.docker.io/sourcegraph/redis-cache:5.3.2@sha256:ed79dada4d1a2bd85fb8450dffe227283ab6ae0e7ce56dc5056fbb8202d95624
150a261
>           successThreshold: 1
157a269
>           protocol: TCP
163,174c275,279
<             - |
<               #!/bin/bash
<               if [ -f /etc/redis/redis.conf ]; then
<                 REDISCLI_AUTH=$(grep -h "requirepass" /etc/redis/redis.conf | cut -d ' ' -f 2)
<               fi
<               response=$(
<                 redis-cli ping
<               )
<               if [ "$response" != "PONG" ]; then
<                 echo "$response"
<                 exit 1
<               fi
---
>             - "\n#!/bin/bash\nif [ -f /etc/redis/redis.conf ]; then\n  REDISCLI_AUTH=$(grep
>               -h \"requirepass\" /etc/redis/redis.conf | cut -d ' ' -f 2)\nfi\nresponse=$(\n
>               \ redis-cli ping\n)\nif [ \"$response\" != \"PONG\" ]; then\n  echo
>               \"$response\"\n  exit 1\nfi\n\t\t\t\t\t"
>           failureThreshold: 3
175a281,283
>           periodSeconds: 10
>           successThreshold: 1
>           timeoutSeconds: 1
187a296
>         terminationMessagePath: /dev/termination-log
192,193c301
<       - env: null
<         image: index.docker.io/sourcegraph/redis_exporter:5.3.2@sha256:21a9dd9214483a42b11d58bf99e4f268f44257a4f67acd436d458797a31b7786
---
>       - image: index.docker.io/sourcegraph/redis_exporter:5.3.2@sha256:21a9dd9214483a42b11d58bf99e4f268f44257a4f67acd436d458797a31b7786
198a307
>           protocol: TCP
209,210c318,320
<           runAsGroup: 1000
<           runAsUser: 999
---
>           runAsGroup: 101
>           runAsUser: 100
>         terminationMessagePath: /dev/termination-log
212c322,324
<       nodeSelector: null
---
>       dnsPolicy: ClusterFirst
>       restartPolicy: Always
>       schedulerName: default-scheduler
216c328,330
<       tolerations: null
---
>         runAsGroup: 101
>         runAsUser: 100
>       terminationGracePeriodSeconds: 30
220a335
> status: {}
222c337
< # helm: Deployment/redis-store
---
> # golden: Deployment/redis-store
227c342,344
<     description: Redis for storing semi-persistent data like user sessions.
---
>     appliance.sourcegraph.com/configHash: 142c00ae31a1d919c6bebf39a02cfd56ecf57e219dc5427b0fbeba316a831ebf
>   creationTimestamp: "2024-04-19T00:00:00Z"
>   generation: 1
229,231c346
<     app.kubernetes.io/component: redis
<     app.kubernetes.io/instance: release-name
<     app.kubernetes.io/managed-by: Helm
---
>     app.kubernetes.io/component: redis-store
233c348
<     app.kubernetes.io/version: 5.3.2
---
>     app.kubernetes.io/version: 5.3.9104
235d349
<     helm.sh/chart: sourcegraph-5.3.2
236a351,360
>   namespace: NORMALIZED_FOR_TESTING
>   ownerReferences:
>   - apiVersion: v1
>     blockOwnerDeletion: true
>     controller: true
>     kind: ConfigMap
>     name: sg
>     uid: NORMALIZED_FOR_TESTING
>   resourceVersion: NORMALIZED_FOR_TESTING
>   uid: NORMALIZED_FOR_TESTING
238a363
>   progressDeadlineSeconds: 600
244,245d368
<       app.kubernetes.io/instance: release-name
<       app.kubernetes.io/name: sourcegraph
251a375
>       creationTimestamp: null
254,255d377
<         app.kubernetes.io/instance: release-name
<         app.kubernetes.io/name: sourcegraph
256a379
>       name: redis-store
258d380
<       affinity: null
260,261c382
<       - env: null
<         image: index.docker.io/sourcegraph/redis-store:5.3.2@sha256:0e3270a5eb293c158093f41145810eb5a154f61a74c9a896690dfdecd1b98b39
---
>       - image: index.docker.io/sourcegraph/redis-store:5.3.2@sha256:0e3270a5eb293c158093f41145810eb5a154f61a74c9a896690dfdecd1b98b39
266a388
>           successThreshold: 1
273a396
>           protocol: TCP
279,290c402,406
<             - |
<               #!/bin/bash
<               if [ -f /etc/redis/redis.conf ]; then
<                 REDISCLI_AUTH=$(grep -h "requirepass" /etc/redis/redis.conf | cut -d ' ' -f 2)
<               fi
<               response=$(
<                 redis-cli ping
<               )
<               if [ "$response" != "PONG" ]; then
<                 echo "$response"
<                 exit 1
<               fi
---
>             - "\n#!/bin/bash\nif [ -f /etc/redis/redis.conf ]; then\n  REDISCLI_AUTH=$(grep
>               -h \"requirepass\" /etc/redis/redis.conf | cut -d ' ' -f 2)\nfi\nresponse=$(\n
>               \ redis-cli ping\n)\nif [ \"$response\" != \"PONG\" ]; then\n  echo
>               \"$response\"\n  exit 1\nfi\n\t\t\t\t\t"
>           failureThreshold: 3
291a408,410
>           periodSeconds: 10
>           successThreshold: 1
>           timeoutSeconds: 1
303a423
>         terminationMessagePath: /dev/termination-log
308,309c428
<       - env: null
<         image: index.docker.io/sourcegraph/redis_exporter:5.3.2@sha256:21a9dd9214483a42b11d58bf99e4f268f44257a4f67acd436d458797a31b7786
---
>       - image: index.docker.io/sourcegraph/redis_exporter:5.3.2@sha256:21a9dd9214483a42b11d58bf99e4f268f44257a4f67acd436d458797a31b7786
314a434
>           protocol: TCP
325,326c445,447
<           runAsGroup: 1000
<           runAsUser: 999
---
>           runAsGroup: 101
>           runAsUser: 100
>         terminationMessagePath: /dev/termination-log
328c449,451
<       nodeSelector: null
---
>       dnsPolicy: ClusterFirst
>       restartPolicy: Always
>       schedulerName: default-scheduler
332c455,457
<       tolerations: null
---
>         runAsGroup: 101
>         runAsUser: 100
>       terminationGracePeriodSeconds: 30
336a462,528
> status: {}
> ---
> # golden: ConfigMap/sg
> apiVersion: v1
> data:
>   spec: |
>     spec:
>       requestedVersion: "5.3.9104"
> 
>       blobstore:
>         disabled: true
> 
>       codeInsights:
>         disabled: true
> 
>       codeIntel:
>         disabled: true
> 
>       frontend:
>         disabled: true
> 
>       gitServer:
>         disabled: true
> 
>       indexedSearch:
>         disabled: true
> 
>       indexedSearchIndexer:
>         disabled: true
> 
>       pgsql:
>         disabled: true
> 
>       postgresExporter:
>         disabled: true
> 
>       preciseCodeIntel:
>         disabled: true
> 
>       redisCache: {}
> 
>       redisStore: {}
> 
>       repoUpdater:
>         disabled: true
> 
>       searcher:
>         disabled: true
> 
>       symbols:
>         disabled: true
> 
>       syntectServer:
>         disabled: true
> 
>       worker:
>         disabled: true
> kind: ConfigMap
> metadata:
>   annotations:
>     appliance.sourcegraph.com/currentVersion: 5.3.9104
>     appliance.sourcegraph.com/managed: "true"
>   creationTimestamp: "2024-04-19T00:00:00Z"
>   name: sg
>   namespace: NORMALIZED_FOR_TESTING
>   resourceVersion: NORMALIZED_FOR_TESTING
>   uid: NORMALIZED_FOR_TESTING

```

File: internal/appliance/spec.go (hunk 1 above)
Comment (jdpleiness) [reply internal/appliance/spec.go:211]: Yeah agreed

**Review by jdpleiness — APPROVED** [review]

**Comment by craigfurman** [comment]
Using this PR to test a new changelog generator tool feature, to exclude PRs based on a no-changelog label.

**Pull request merged**